contents/handbook/engineering/on-call-rotation.md

```diff
@@ -0,0 +1,39 @@
+---
+title: On-call rotation
+sidebar: Handbook
+showTitle: true
+---
+
+Every week, three engineers are on call in shifts. If you're one of the lucky three, congratulations!
+
+PostHog Cloud doesn't shut down at night (_whose_ night anyway?) nor on Sunday. As a 24/7 service, our goal is to be 100% operational 100% of the time. The universe doesn't always agree and hiccups do occur, which we catch in the form of service alerts. To make sure no alert goes unhandled, there is **always** one engineer on call, tasked with first-line defense.
+
+## When is my turn?
+
+We're very lucky that we have an almost-follow-the-sun rotation for service alerts, so nobody gets woken up in the middle of the night. Every engineer at PostHog is expected to be part of the rotation within one of the three PagerDuty schedules:
+
+- [Europe](https://posthog.pagerduty.com/schedules#PF7ZGBT) (08:00 to 16:00 UTC)
+- [Americas East](https://posthog.pagerduty.com/schedules#PW1E9Y4) (16:00 to 00:00 UTC)
+- [Americas West](https://posthog.pagerduty.com/schedules#P3J10CZ) (00:00 to 08:00 UTC)
+
+Before going on call, make sure you have the [PagerDuty mobile app](https://support.pagerduty.com/docs/mobile-app) installed and configured. This way it'll be harder to miss an alert.
+
+## What if I'm scheduled for a week when I won't be available?
+
+Swap with another engineer in advance! Find a volunteer by asking in Slack, then use PagerDuty schedule overrides. You can trade whole weeks, but also just specific days. Remember not to alter the rotation's core order, as that's an easy way to accidentally shift the schedule for everyone.
+
+## What do I do on call?
+
+You just go about your day, paying no attention to being on call most of the time! But keep your laptop at hand, because when you do get paged, it's time to switch gears immediately and start investigating the alert.
+
+A chunk of our high-frequency alerts have runbooks attached, most of which live [on the internal runbooks site](http://runbooks/). The runbook should tell you what to look at and easy fixes if there are any. Every alert also has a link to the Grafana graph that triggered the alert.
+
+[If it looks like the alert should be raised as a public incident, go raise an incident.](https://posthog.com/handbook/engineering/incidents)
+
+If you're stumped and no resource is of help, get someone from the relevant team to shadow you while you sort the problem out. The idea is that they can help you understand the issue and where to find how to debug it. The idea is _not_ for them to take over at this point, as otherwise you won't be able to learn from this incident.
```

Comment:
this is actually where we: at least pipeline and infra would like you to use the secondary oncall as the random person to pick.

Realistically what happens is you escalate and you work together. They might even say they got it & take over - that's fine, BUT the on-call should be responsible for making sure we always leave the campsite cleaner than we found it and add some more steps for this alert, that the next on-call could do before escalating. If there aren't any runbooks, add it and first 3 steps, then next time someone else can add another step or two that can be done before escalating. This is a very low time & effort way to get great runbooks / on-call sanity in a surprisingly short time as we're improving the runbooks for things that actually happen & history is the best indicator of the future. TLDR: if you escalated you should improve the runbook to make the next oncall's probability to need to escalate lower

Reply:
I think there is a point to secondary on-call, but it's not maintained at all. There are three on-call schedules distributed geographically, yet only a single secondary one, which immediately makes it useless ([schedules here](https://posthog.pagerduty.com/schedules-new)). As an example for the current week, it makes zero, none, null sense to wake @mariusandra up in the middle of the night for an escalated ingestion delay – yet Marius _is_ currently on the hook.
If you're up for it though, I totally support you @tiina303 if you want to rethink secondary on-call. I think it's worth doing that separately from this revision

Reply:
I agree for the not being awake part and if there are several people around just posting in the team channel for help could be a good solution - the team can self organise to have the current secondary on-call jump in if they are awake

Reply:
Based on what I noted above, I removed the existing all-team secondary schedule, so that we don't wake Marius on accident